sherpa/plot/__init__.py

```diff
@@ -113,11 +116,91 @@
            'Confidence1D', 'Confidence2D',
            'IntervalProjection', 'IntervalUncertainty',
            'RegionProjection', 'RegionUncertainty',
+           'TemporaryPlottingBackend',
+           'set_backend',
            )
 
 _stats_noerr = ('cash', 'cstat', 'leastsq', 'wstat')
 
 
+def set_backend(name):
+    '''Set the Sherpa plotting backend.
+
+    Plotting backends are registered in Sherpa with a string name.
+    See the examples below for how to get a list of available backends.
+
+    Parameters
+    ----------
+    name : string
+        Name of the plotting backend.
+
+    Example
+    -------
+    Set the backend to use Pylab from matplotlib for plotting. This is
+    probably what most users need:
+
+        >>> from sherpa.plot import set_backend
+        >>> set_backend('pylab')
+
+    Get a list of registered backends:
+
+        >>> from sherpa.plot import PLOT_BACKENDS
+        >>> PLOT_BACKENDS
+        {'BaseBackend': sherpa.plot.backends.BaseBackend,
+         'BasicBackend': sherpa.plot.backends.BasicBackend,
+         'IndepOnlyBackend': sherpa.plot.backends.IndepOnlyBackend,
+         'pylab': sherpa.plot.pylab_backend.PylabBackend}
+
+    This list shows the names and the class for each backend.
+    Details for each backend can be found in the Sherpa documentation or by
+    inspecting the backend classes using normal Python functionality:
+
+        >>> from sherpa.plot.backends import BasicBackend
+        >>> help(BasicBackend)
+    '''
+    global backend
+    if name in PLOT_BACKENDS:
+        backend = PLOT_BACKENDS[name]()
+    else:
+        warning(f'Plotting backend not changed because {name} is not a known plotting backend.\n' +
+                f'List of available plotting backends: {list(PLOT_BACKENDS.keys())}',
+                )
+
+
+# TODO: make real example
+class TemporaryPlottingBackend(contextlib.AbstractContextManager):
+    '''Set the Sherpa plotting backend as a context, e.g. for a single plot
+
+    This changes the logging level globally for all modules in sherpa.
+
+    Parameters
+    ----------
+    backend : object
+        Set a sherpa plotting backend. The backend can be passed in as an
+        object, or as a convenience, as a simple string
+
```

Comment:
I think that `backend` must be a string now, given how `set_backend` works. Is that right?